cli/jhipster.js

```diff
@@ -35,21 +35,31 @@ if (!semver.satisfies(currentNodeVersion, minimumNodeVersion)) {
     process.exit(1);
 }
 
-let preferGlobal = false;
-if (process.argv.length >= 3) {
-    if (process.argv[2] === 'upgrade') {
-        // Prefer global version for `jhipster upgrade` to get most recent code
-        preferGlobal = true;
-    }
-}
-requireCLI(preferGlobal);
+let preferLocal = true;
+
+program.command('upgrade')
```

Comment:
I don't think so because I tested it (not all generators but upgrade and app are fine) and because program is defined as a const in both jhipster.js and cli.js so if they were interfering I would expect cli to raise an error for assigning twice a constant. I will check under debugger anyway.